decompiler/frontend/binaryninja/handlers/assignments.py

```diff
@@ -84,6 +85,10 @@ def lift_get_field(self, instruction: mediumlevelil.MediumLevelILVarField, is_al
 
     def lift_store(self, assignment: mediumlevelil.MediumLevelILStoreSsa, **kwargs) -> Assignment:
         """Lift a store operation to pseudo (e.g. [ebp+4] = eax)."""
+        op = self._lifter.lift(assignment.dest, parent=assignment)
+        if isinstance(op, GlobalVariable):
+            op.ssa_label = assignment.dest_memory
+            return Assignment(op, self._lifter.lift(assignment.src), writes_memory=assignment.dest_memory)
```

Comment:
I will refactor that.

However, I do not completely agree with the point about `ssa_label`. First, aliased variables are allowed to have ssa labels. Second, later comment suggests that having `wites_memory` property for an assignment is an unoptimal solution. But in order missing definitions work with globals, we need either to keep track on memory version by setting correct ssa label of global variable or by setting `writes_memory` property of the assignment. Otherwise, we are missing important infos. Nevertheless, setting both could be actually an overkill.

Reply:
Additionally, could you please elaborate why setting _public_ field of an object is a _side effect_?

Reply:
Refactored.